src/charm.py

```diff
@@ -75,14 +76,15 @@ def _on_config_changed(self, event):
         logger.debug("Running config changed handler")
         container = self.unit.get_container("mongodb")
 
-        # Set security key
-        try:
-            if not self.have_security_key(container):
-                self.set_security_key(container)
-        except ConnectionError:
+        if not container.can_connect():
             self.unit.status = WaitingStatus("Waiting for Pebble ready")
+            event.defer()
```

Comment:
Defer does not eliminate to possibility that the charm would have to wait for update_status.
Unless we now know for sure that during startup, config_changed is guaranteed to fire before other events (pebble_ready, start). IIRC, during early work on alertmanager I occasionally observed config_changed firing last.

Reply:
Agreed. I will remove the defer() and retest. The Problem I was seeing with Juju is no longer reproducible so I can not be sure if this defer would solve any problem.

Reply:
Removed defer and retested (using mongoconsumer).